main/src/ca/uwaterloo/flix/language/phase/Codegen.scala

```diff
@@ -704,30 +698,7 @@ object Codegen {
     case Type.Native => // Don't need to cast AnyRef to anything
 
     case Type.Apply(Type.FTuple(l), _) =>
-      // This is actually a bit more complicated, since we have multiple representations for a tuple (e.g. Value.Tuple,
-      // Array, scala.TupleN). We have to call `Value.cast2tuple` instead of doing a direct cast.
-
-      // Load the Value singleton object. Do a SWAP to put stack operands in the right order, then call
-      // `Value.cast2tuple`, to get the elements array.
-      Constants.loadValueObject(visitor)
-      visitor.visitInsn(SWAP)
-      visitor.visitMethodInsn(INVOKEVIRTUAL, Constants.valueObject, "cast2tuple", "(Ljava/lang/Object;)[Ljava/lang/Object;", false)
-
-      // TODO: Update this when we remove Value.Tuple.
-      // cast2tuple returns an Array, but we expect a Value.Tuple. So we have to construct one.
-      // Create the Value.Tuple reference.
-      visitor.visitTypeInsn(NEW, asm.Type.getInternalName(Constants.tupleClass))
-
-      // We use dup_x1 and swap to manipulate the stack so we can avoid using a local variable.
-      // Stack before: array, tuple (top)
-      // Stack after: tuple, tuple, array (top)
-      visitor.visitInsn(DUP_X1)
-      visitor.visitInsn(SWAP)
-
-      // Finally, call the constructor, which pops the reference (tuple) and argument (array).
-      val clazz = Constants.tupleClass
-      val ctor = clazz.getConstructor(Constants.arrayObjectClass)
-      visitor.visitMethodInsn(INVOKESPECIAL, asm.Type.getInternalName(clazz), "<init>", asm.Type.getConstructorDescriptor(ctor), false)
+      visitor.visitTypeInsn(CHECKCAST, "[Ljava/lang/Object;")
```

Comment:
I would actually move this case into the one covered by line 686. That way, the (outer) match has three cases: primitives (which need to be unboxed), reference types (which need a `CHECKCAST`), and `Type.Native` (which does nothing).